### PR TITLE
Fix CodyInstalled logic for relogins

### PR DESCRIPTION
**Pull request by nathan-downs**
## Test plan

Run VS Code extension in debug mode
Verify events in event_logs
Verify events on dotcom

### CODE REVIEW

**Comment by nathan-downs** [comment]
Screenshot 2023-05-23 at 7 02 38 PM

File: client/cody-shared/src/telemetry/EventLogger.ts [hunk 1]
```diff
@@ -37,7 +37,6 @@ export class EventLogger {
             serverEndpoint: this.serverEndpoint,
             extensionDetails: this.extensionDetails,
         }
-        console.log('api requqest for anonID: ' + anonymousUserID)
```
Comment (nathan-downs) [inline client/cody-shared/src/telemetry/EventLogger.ts:40]: Old debug line.

File: client/cody/src/event-logger.ts [hunk 2]
```diff
@@ -38,7 +42,7 @@ export function logEvent(eventName: string, eventProperties?: any, publicPropert
         ...publicProperties,
         version: packageVersion,
     }
-    void eventLogger.log(eventName, anonymousUserID, argument, publicArgument)
+    await eventLogger.log(eventName, anonymousUserID, argument, publicArgument)
```
Comment (nathan-downs) [inline client/cody/src/event-logger.ts:45]: This is the second part of what was breaking event logging.

**Comment by nathan-downs** [comment]
Screenshot 2023-05-23 at 8 43 56 PM

**Comment by nathan-downs** [comment]
Verified that this is fixed and showing up both in the s2 event_logs and in dotcom as an anonymous event.

File: client/cody/src/completions/index.ts [hunk 3]
```diff
@@ -221,13 +221,13 @@ export class CodyCompletionItemProvider implements vscode.InlineCompletionItemPr
             return []
         }
 
-        logEvent('CodyVSCodeExtension:completion:started', LOG_INLINE, LOG_INLINE)
+        await logEvent('CodyVSCodeExtension:completion:started', LOG_INLINE, LOG_INLINE)
```
Comment (philipp-spiess) [inline client/cody/src/completions/index.ts:224]: Do we need to await the event sending everywhere? This introduces additional latency to some of the highly interacted with code paths for something that is not needed for the user to know about. 

If there are issues with out-of-order events we should implement proper queueing on the event logger instance instead.

```suggestion
        logEvent('CodyVSCodeExtension:completion:started', LOG_INLINE, LOG_INLINE)
```

File: client/cody/src/event-logger.ts (hunk 2 above)
Comment (philipp-spiess) [reply client/cody/src/event-logger.ts:45]: Can you clarify how this is breaking event logging? I think the only difference here is that we do not wait for the event to be sent before completing with the rest of the logic

**Review by philipp-spiess — CHANGES_REQUESTED** [review]
I think we need some changes here. This logic starts to await network roundtrips (up to two requests per event) in very latency susceptible parts of the code base (e.g. completion calculation).

It's unclear to me why this is needed. Do we have issues with out-of-order events? If so, can we add a queue to the event logger interface instead?

File: client/cody/src/completions/index.ts (hunk 3 above)
Comment (nathan-downs) [reply client/cody/src/completions/index.ts:224]: I'll do some testing on removing that--logEvent is making an API call that wasn't getting completed that I'm trying to solve for

**Comment by nathan-downs** [comment]
Screenshot 2023-05-24 at 9 07 18 PM
Screenshot 2023-05-24 at 9 07 09 PM

**Review by philipp-spiess — APPROVED** [review]
Nice! Thank you for changing this to not block our critical paths 💪

**Pull request merged**